src/main/java/zd/zero/waifu/motivator/plugin/settings/WaifuMotivatorSettingsPage.java

```diff
@@ -189,6 +203,8 @@ public void apply() {
         ApplicationManager.getApplication().getMessageBus()
             .syncPublisher( PluginSettingsListener.Companion.getPLUGIN_SETTINGS_TOPIC() )
             .settingsUpdated( this.state );
+
+        motivationStatistics.resetStatistics();
```

Comment:
> I was curious as to why you didn't add a plugin settings topic listener in motivation statistics.

For this, I'm still having a hard time when or when not to use topic listeners, should I always go for a topic publisher and consumer whenever there's an opportunity?

Reply:
So you're suggesting is to subscribe to this topic instead right?
```
        ApplicationManager.getApplication().getMessageBus()
            .syncPublisher( PluginSettingsListener.Companion.getPLUGIN_SETTINGS_TOPIC() )
            .settingsUpdated( this.state );
```

Reply:
> So you're suggesting is to subscribe to this topic instead right

Yeah, I like it because it keeps the relevant code together and prevent's this file from becoming a hot-spot for more code not related to settings ui stuff.